CONTRIBUTING.md

```diff
@@ -1,136 +1,121 @@
+change in the proposal, please start by creating a new issue in the tracker.
+Someone should respond within 48 hours.  If the proposal is accepted, then you
+can follow the process below to begin work on it.
+
+- Finally, if you have noticed an obvious typo or bug that can be fixed with a
+small code change, please feel free to submit a PR directly addressing the issue
+without opening an issue.
+
+## Build Process
+
+Nearcore is a reasonably standard Rust project, so `cargo test` most likely will
+just work.  There are couple of specifics though:
+
+* `nearcore` assumes a UNIX-like environment.  Linux is actively supported; Mac
+is generally supported (the newer M1 chipset may not be well supported); and
+Windows is not supported.
+* `nearcore` build process includes generating bindings to RocksDB via
+`bindgen`, which requires `libclang`.  See [`bindgen`
+documentation](https://rust-lang.github.io/rust-bindgen/requirements.html#clang)
+for installation instructions.
+* You can optionally use the system installations of `librocksdb`, `libsnappy`
+and `lz4` in order to speed up the compilation and reduce the build memory
+requirements by setting the `ROCKSDB_LIB_DIR`, `SNAPPY_LIB_DIR` and
+`LZ4_LIB_DIR` environment variables.  These environment variables should point
+at the directory containing the dynamic shared objects (`.so`s or `.dylib`s).
+Please see the
+[wiki](https://wiki.near.org/contribute/contribute-nearcore#use-system-provided-rocksdb)
+for additional instructions on how to use a system-provided rocksdb
+installation.
+
+If your setup does not involve `rustup`, the required version of the rust
+toolchain can be found in the `rust-toolchain` file.
+
+## Pull Requests
+
+All the contributions to `nearcore` happen via Pull Requests.  To create a Pull
+Request, fork `nearcore`, create a new branch, do the work there, and then send
+a PR against the `master` branch.
+
+When working on a PR, please keep the following in mind:
+
+1. The changes should be thoroughly tested.  Please refer to [this
+document](https://github.com/nearprotocol/nearcore/wiki/Writing-tests-for-nearcore)
+for our testing guidelines and overview of the testing infrastructure.
+2. Feel free to submit draft PRs to get early feedback and to make sure you are
+on the right track.
+3. A PR can contain any number of commits and when merged, the commits will be
+squashed into a single commit.
+4. The PR name should follow the template: `<type>: <name>`.  
+Where `type` is:
+   - `fix` for bug fixes;
+   - `feat` for new features;
+   - `refactor` for changes that reorganize code without adding new content;
+   - `doc` for changes that change documentation or comments;
+   - `test` for changes that introduce new tests;
+   - `chore` for grunt tasks like updating dependencies.
+5. The PR description should follow the following template:
 
 ```
-<type>: <name>
-
 <description>
 
 Test plan
 ---------
 <test plan>
 ```
```

Comment:
I'd want to bike shed on whether we need this rule or not. We are definitely not following it, so we should either start enforcing it, or to remove it. But this is out of scope for the PR.

Reply:
I was also suspicious of this.

Reply:
Agreed, will remove.